src/blog/2025/10/plc-to-mqtt-using-flowfuse.md

```diff
@@ -0,0 +1,187 @@
+
+![Change node adding context to data](./images/change-node.png){data-zoomable}
+*Change node used to convert temperature and add machine context*
+
+**Function Nodes for Custom Logic**
+
+Function nodes provide full JavaScript access for complex requirements. Write custom logic, install npm packages, and access the complete JavaScript standard library when Change nodes and JSONata reach their limits.
+
+*Tip: Use the [FlowFuse assistant](/blog/2025/07/flowfuse-ai-assistant-better-node-red-manufacturing/) to generate function nodes. Describe the transformation you need in plain English, and it will create the code for you. For best results, provide sample input data to ensure the output matches your requirements.*
+
+**Pre-built Community Nodes**
+
+Before building custom solutions, check the palette manager. The Node-RED ecosystem includes thousands of nodes for data aggregation, statistical analysis, time-series buffering, and unit conversions. Many common transformation tasks already have ready-made solutions.
+
+For example, a popular node I’m using in my demo for parsing and transforming data is `node-red-contrib-buffer-parser`. This node is especially useful when working with Modbus or PLC outputs, as it converts raw buffers into structured formats that can be easily processed further.
+
+## Step 3: Set Up MQTT with FlowFuse
+
+Most MQTT implementations require setting up a separate broker either paying for a managed service or hosting your own. FlowFuse includes a managed MQTT broker built directly into the platform, eliminating this extra step.
+
+Traditional PLC-to-cloud setups typically involve several moving parts: edge gateways running protocol drivers, a separate MQTT broker (cloud-hosted or self-managed), and your destination cloud services. Each layer adds configuration work, licensing costs, and potential failure points. When data stops flowing, you end up troubleshooting across multiple systems to locate the issue.
+
+FlowFuse consolidates those layers into a single integrated platform. It provides enterprise-grade features for management, scaling, deployment, and security—all handled by the FlowFuse infrastructure. You retain full control over configuration settings through a clean, intuitive interface, without needing to maintain multiple external systems.
+
+To use the FlowFuse MQTT broker, you’ll need a FlowFuse Pro or higher-tier account. Once on the Pro plan, you can enable the managed MQTT service by navigating to the Broker section from the left sidebar and selecting FlowFuse Broker.
+
+![Enabling FlowFuse MQTT Broker](./images/set-broker.png){data-zoomable}
+_Enabling FlowFuse MQTT Broker_
+
+**Configure Publishing**
+
+1. Drag a **FlowFuse MQTT Out** node onto your canvas.
+2. Open the node configuration. You do not need to manually configure the node — it will automatically pick up its configuration.
+3. Set the topic to something like:
+
+```
+plant_a/assembly/line_1/temperature
```

Comment:
Observation: this suggested format differs from the screenshots and video 

Also, I would recommend sticking to formats proposed in [best  practices](https://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices/) e.g.

`company/site/area/line/cell/device/measurement`

Note no underscores, starts at company level (for when companies merge or cooperate like nissan + renault or sub companies like alphabet has google and nest),